tinygrad/helpers.py

```diff
@@ -1,18 +1,23 @@
 from __future__ import annotations
-import platform
-from dataclasses import dataclass, asdict
-import os, math, functools, time, re
+import os, functools, platform, time
+from weakref import KeyedRef, ref
+from _weakref import _remove_dead_weakref
 import numpy as np
-from typing import Tuple, Union, List, NamedTuple, Final, Iterator, ClassVar, Optional, Callable, Any
+from typing import Dict, Tuple, Union, List, NamedTuple, Final, Iterator, ClassVar, Optional, Callable, Any
+from math import prod
+
 ShapeType = Tuple[int, ...]
 # NOTE: helpers is not allowed to import from anything else in tinygrad
 OSX = platform.system() == "Darwin"
 
 def dedup(x): return list(dict.fromkeys(x))   # retains list order
-def prod(x:Union[List[int], Tuple[int, ...]]) -> int: return math.prod(x)
-def argfix(*x): return tuple() if len(x) == 0 else tuple(x[0]) if isinstance(x[0], (tuple, list)) else tuple(x)
+def argfix(*x): 
+  if x[0].__class__ in {tuple, list}:
+    try: return tuple(x[0])
+    except IndexError: return tuple()
+  return tuple(x)
 def argsort(x): return type(x)(sorted(range(len(x)), key=x.__getitem__)) # https://stackoverflow.com/questions/3382352/equivalent-of-numpy-argsort-in-basic-python
-def all_same(items): return all(x == items[0] for x in items) if len(items) > 0 else True
+def all_same(items): return all([x == items[0] for x in items]) if len(items) > 0 else True
```

Comment:
`def all_same(items): return all([x == items[0] for x in items]) if len(items) > 1 else True`

Reply:
Correct - however, almost all lists that go in here are <10 items (mostly 1 or 2) long.
In practice, all([]) and any([]) is faster in this codebase based on profiling the calls.